client/src/main/java/io/pravega/client/admin/impl/StreamManagerImpl.java

```diff
@@ -141,12 +141,27 @@ public boolean checkStreamExists(String scopeName, String streamName) {
     }
 
     @Override
+    @Deprecated
     public boolean deleteScope(String scopeName) {
+        return deleteScope(scopeName, false);
+    }
+
+    @Override
+    public boolean deleteScope(String scopeName, boolean deleteStreams) {
         NameUtils.validateUserScopeName(scopeName);
         log.info("Deleting scope: {}", scopeName);
-        return  Futures.getThrowingException(controller.deleteScope(scopeName));
+        
+        if (deleteStreams) {
+            Iterator<Stream> iterator = listStreams(scopeName);
+            while (iterator.hasNext()) {
+                Stream stream = iterator.next();
+                Futures.getThrowingException(controller.sealStream(stream.getScope(), stream.getStreamName()));
```

Comment:
If a "seal" or "deleteStream" operation fails for a single stream should'nt we continue deleting the other streams?

Reply:
Well, even if we opportunistically delete other streams, the scope can not be deleted if a single stream's seal and delete failed. So we should throw the error back to the user so that they can decide on how to handle it. 
a subsequent attempt to delete the scope using this api will iterate over all available streams and attemp to delete them.